lib/vintage_net_qmi/connectivity.ex

```diff
@@ -158,33 +200,102 @@ defmodule VintageNetQMI.Connectivity do
 
   def handle_info(_message, state), do: {:noreply, state}
 
+  defp update_derived_status(state) do
+    new_derived_status = derive_status(state)
+
+    new_grace_timer =
+      case new_derived_status do
+        :going_down ->
+          state.grace_timer || schedule_grace_timer()
+
+        _ ->
+          if state.grace_timer do
+            :timer.cancel(state.grace_timer)
```

Comment:
Should there be a call to `schedule_grace_timer/0` here? If not, this case should probably return nil -- otherwise we'll end up with `{:ok, :cancel}` (or potentially an error) in state.

Reply:
Good catch. Also, I need a comment here to explain this case.